src/main/java/cn/nukkit/Server.java

```diff
@@ -347,6 +347,10 @@ public Level remove(Object key) {
     @Since("1.19.50-r1")
     private FreezableArrayManager freezableArrayManager;
 
+    @PowerNukkitXOnly
+    @Since("1.19.70-r3")
+    public Pattern playerNameRegex = Pattern.compile("^(?![_ ])([a-zA-Z0-9_]{3,16}(?: [a-zA-Z0-9_]{3,17})*)(?<![_ ])$");
```

Comment:
Now names with a space in the middle like `Joe Biden` are not allowed, however it should be.
Maybe we lost a space in the charset `[a-zA-Z0-9_]`?

Reply:
Name with space in it perfectly works, my nickname is "koshak mine" and i can join

Reply:
Sorry for giving an incorrect example earlier. It seems that a two character surname is not allowed. But this is at least common in Chinese names translated into English. Can you make some modifications to this? esteem it a favor

Reply:
If I understand you correctly, it should work now

Reply:
thanks, LGTM.
I'm requesting for at least one more review, if everything is ok, this PR will be merged.